### PR TITLE
Feature/configure event frequencies

### PR DESCRIPTION
**Pull request by colmsnowplow**

### CODE REVIEW

**Comment by colmsnowplow** [comment]
@istreeter this one I'm not as sure about so not asking for review but keen for feedback (sorry for spamming PRs, this one's adding to your work so feels natural to ask you first).

I've added similar frequencies functionality for the distribution of event types, and I've made both configurable.

I'm not 100% certain about how I've done this since a) it means config now depends on some of the sub packages, and b) it's slightly different from how config options are passed elsewhere, and c) long time since I've worked with scala patterns and this is a lot of LOC.

In my head this is a decent quick implementation that doesn't require refactoring. But I don't know if it makes for a good enough implementation that we'd be happy to merge it (wouldn't be a problem if not! I have all the features I need for my load testing, what does/doesn't make it into a release isn't a hard blocker.

**Review by istreeter — APPROVED** [review]
I love this!

This would have been helpful for me recently when I needed to produce an output file containing unstruct events only.  I achieved it that time by making a temporary change to the code, but then it was impossible to document how I generated the file.  This PR is perfect because it makes it configurable.

> a) it means config now depends on some of the sub packages

This is fine, we often do this.

> b) it's slightly different from how config options are passed elsewhere

You'll need to explain this one to me -- it looks very similar to how config options are passed elsewhere.

> c) ... this is a lot of LOC.

It isn't 😉

**Comment by colmsnowplow** [comment]
> You'll need to explain this one to me -- it looks very similar to how config options are passed elsewhere.

After a weekend coming back to it - I'm not sure what I meant. I made this PR while trying to move a few other things forward at once, so I think I was just confuesd :D

>>  c) ... this is a lot of LOC.

> It isn't 😉

You're right it isn't. I dunno what I was thinking tbh - I think because a lot of functions needed to be changed it was big in my head 🤷

File: sinks/src/main/scala/com.snowplowanalytics.snowplow.eventgen/Main.scala [hunk 1]
```diff
@@ -53,15 +53,18 @@ object Main extends IOApp {
                     dups.synTotal,
                     config.eventPerPayloadMin,
                     config.eventPerPayloadMax,
-                    time
+                    time,
+                    // TODO: Add to config (along with unstructs) and parse properly
```
[the remark this answers is not in the document]
Comment (colmsnowplow) [reply sinks/src/main/scala/com.snowplowanalytics.snowplow.eventgen/Main.scala:57]: 🤦 I need to stop leaving these in - yeah I hardcoded some things to test the concept, and forgot to  remove the todo later.

**Pull request merged**